src/server/handlers/launchpad.js

```diff
@@ -435,6 +437,30 @@ export const findSnap = (req, res) => {
     .catch((error) => sendError(res, error));
 };
 
+const clearSnapCache = (repositoryUrl) => {
+  const repository = parseGitHubUrl(repositoryUrl);
+  const enabledReposCacheId = getUrlPrefixCacheId(getRepoUrlPrefix(repository.owner));
+  const snapCacheId = getRepositoryUrlCacheId(repositoryUrl);
+  const snapNameCacheId = getSnapNameCacheId(repositoryUrl);
+
+  logger.info(`Clearing caches for ${repositoryUrl}: ${enabledReposCacheId}, ${snapCacheId}, ${snapNameCacheId}`);
+  getMemcached().del(enabledReposCacheId, (err) => {
+    if (err) {
+      logger.error(`Error deleting ${enabledReposCacheId} from memcached:`, err);
+    }
+  });
+  getMemcached().del(snapCacheId, (err) => {
+    if (err) {
+      logger.error(`Error deleting ${snapCacheId} from memcached:`, err);
+    }
+  });
+  getMemcached().del(snapNameCacheId, (err) => {
+    if (err) {
+      logger.error(`Error deleting ${snapNameCacheId} from memcached:`, err);
+    }
+  });
```

Comment:
I didn't want to delay the response on memcached calls, but yes - in theory this could fail in tests (as memcached may be cleared after response is returned).